### PR TITLE
remove `options.resolve` and add `render`-callback instead

### PR DESCRIPTION
**Pull request by mindplay-dk**
As asked in #55, let's drop `options.resolve` and add the `render`-callback instead, like React.

Instead of passing it through many functions, I simply attached the `done` callback to the fiber itself - and then check for the `done`-callback in `commitWork` at the end, and call it.

Does this seem correct to you?

The test is passing, so hopefully this was as easy as it appears to be? 🙂

### CODE REVIEW

File: src/reconciler.js [hunk 1]
```diff
@@ -64,7 +65,7 @@ function updateHost (WIP) {
 
 function getParentNode (fiber) {
   if (!fiber.parent) return fiber.node
-  while (fiber.parent.tag === HOOK) return fiber.parent.parent.node
+  if (fiber.parent.tag === HOOK) return fiber.parent.parent.node
```
Comment (mindplay-dk) [inline src/reconciler.js]: Note that this change is unrelated to the PR - maybe this was a `while` statement in an earlier version, but an `if` statement should be enough, since all it does it `return` unconditionally?

Comment (132yse) [reply src/reconciler.js]: Here still need a while loop, because the parent of the component may still be the component

Comment (mindplay-dk) [reply src/reconciler.js]: But this `while`-statement doesn't loop.

The body of the `while`-statement is an unconditional `return`-statement - so if the condition in the `while`-statement evaluates true, it returns; exactly like an `if`-statement.

Comment (132yse) [reply src/reconciler.js]: Oh, you're right. It's my fault. I'll fix it.

**Comment by 132yse** [comment]
I fixed `getParentNode` bug just now :happy:
